pre-commit

PLangowski opened the pull request.

DaniilKl commented on meta-dts-distro/recipes-support/cloudsend/cloudsend_git.bb:
There seem to be a lot of changes between versions `2.2.1` and `2.2.8`:
```diff
diff --git a/home/danillklimuk/Downloads/cloudsend.sh b/cloudsend.sh
old mode 100644
new mode 100755
index f9376c4b85cc..71892acd1cbb
--- a/home/danillklimuk/Downloads/cloudsend.sh
+++ b/cloudsend.sh
@@ -21,9 +21,9 @@
 # Get this script to current folder with:
 # curl -O 'https://raw.githubusercontent.com/tavinus/cloudsend.sh/master/cloudsend.sh' && chmod +x cloudsend.sh
 #
-# NOTE: Cloudsend 2 changed the way password is handled and
+# NOTE: Cloudsend 2 changed the way password is handled and 
 # is NOT compatible with cloudsend 1 calls. The -e parameter
-# now does what -p parameter did (environment passwords),
+# now does what -p parameter did (environment passwords), 
 # while the -p parameter receives the password directly.
 #
 ############################################################
@@ -32,13 +32,14 @@
 
 
 
-CS_VERSION="2.2.1"
+CS_VERSION="2.2.8"
 
 TRUE=0
 FALSE=1
 
 CLOUDURL=""
 FOLDERTOKEN=""
+INNERPATH=""
 
 PUBSUFFIX="public.php/webdav"
 HEADER='X-Requested-With: XMLHttpRequest'
@@ -50,6 +51,9 @@ OUTFILE=''
 RENAMING=$FALSE
 QUIETMODE=$FALSE
 GLOBBING=$FALSE
+LIMITTING=$FALSE
+LIMITCMD=''
+RATELIMIT=''
 GLOBCMD=' -g'
 VERBOSE=' --progress-bar'
 
@@ -62,6 +66,8 @@ DIRLIST=()
 FILELIST=()
 
 CURLEXIT=0
+CURLRESPONSES=""
+ABORTONERRORS=$FALSE
 
 
 
@@ -116,6 +122,15 @@ curlAddExitCode() {
 }
 
 
+# Curl appended messages
+# Will probably be empty if curl was able to perfom as intended
+curlAddResponse() {
+        if isNotEmpty "$1"; then
+                isEmpty "$CURLRESPONSES" && CURLRESPONSES="$2"$'\n'"$1" || CURLRESPONSES="$CURLRESPONSES"$'\n----------------\n'"$2"$'\n'"$1"
+        fi
+}
+
+
 # Prints usage information (help)
 usage() {
         printVersion
@@ -127,6 +142,8 @@ Parameters:
   -r | --rename <file.xxx> Change the destination file name
   -g | --glob              Disable input file checking to use curl globs
   -k | --insecure          Uses curl with -k option (https insecure)
+  -l | --limit-rate        Uses curl limit-rate (eg 100k, 1M)
+  -a | --abort-on-errors   Aborts on Webdav response errors
   -p | --password <pass>   Uses <pass> as shared folder password
   -e | --envpass           Uses env var \$CLOUDSEND_PASSWORD as share password
                            You can 'export CLOUDSEND_PASSWORD' at your system, or set it at the call
@@ -170,7 +187,7 @@ Examples:
   ./cloudsend.sh './myfile.txt' 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28'
   ./cloudsend.sh 'my Folder' 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28'
   ./cloudsend.sh -r 'RenamedFile.txt' './myfile.txt' 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28'
-  ./cloudsend.sh -p 'MySecretPass' './myfile.txt' 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28'
+  ./cloudsend.sh --limit-rate 200K -p 'MySecretPass' './myfile.txt' 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28'
   ./cloudsend.sh -p 'MySecretPass' -r 'RenamedFile.txt' './myfile.txt' 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28'
   ./cloudsend.sh -g -p 'MySecretPass' '{file1,file2,file3}' 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28'
   cat file | ./cloudsend.sh - 'https://cloud.mydomain.net/s/fLDzToZF4MLvG28' -r destFileName
@@ -233,8 +250,18 @@ parseOptions() {
                         -g|--glob)
                                 GLOBBING=$TRUE
                                 GLOBCMD=''
-                                log "> Glob mode on, input file checkings disabled"
+                                log "> Glob mode ON, input file checkings disabled"
+                                shift ;;
+                        -a|--abort-on-errors)
+                                ABORTONERRORS=$TRUE
+                                log "> Abort on errors ON, will stop execution on DAV errors"
                                 shift ;;
+                        -l|--limit-rate)
+                                loadLimit "${2}"
+                                LIMITTING=$TRUE
+                                log "> Rate limit set to $RATELIMIT"
+                                shift ; shift ;;
+                                
                         *)
                                 if isEmpty "$1"; then
                                         break ;
@@ -242,10 +269,10 @@ parseOptions() {
                                         CLOUDSEND_PARAMS=("${CLOUDSEND_PARAMS[@]}" "$1")
                                         shift ;
                                 fi
-
+                                        
                 esac
         done
-
+        
         CLOUDURL=''
         FILENAME="${CLOUDSEND_PARAMS[0]}"
         CLOUDSHARE="${CLOUDSEND_PARAMS[1]}"
@@ -257,8 +284,18 @@ parseOptions() {
                 CLOUDURL="${CLOUDSHARE%/s/*}"
         fi
 
+        # get token and sub folder
         FOLDERTOKEN="${CLOUDSHARE##*/s/}"
+        INNERPATH="${FOLDERTOKEN##*\?path=}"
+        FOLDERTOKEN="${FOLDERTOKEN%\?*}"
+        
+        if [[ "$FOLDERTOKEN" == "$INNERPATH" ]]; then
+                INNERPATH=""
+        else
+                INNERPATH="$(decodeSlash "$INNERPATH")"
+        fi
 
+        
         if isGlobbing; then
                 if isRenaming; then
                         initError $'Cannot rename output files when using globbing on input.\nAll files would get the same output name and then be overwritten.\nSend individual files if you need renaming.'
@@ -285,6 +322,15 @@ parseOptions() {
         log ''
 }
 
+# Parses Rate limitting
+loadLimit() {
+        if [ -z "$@" ]; then
+                initError "Trying to set an empty rate limit"
+        fi
+        RATELIMIT="$@"
+        LIMITCMD=' --limit-rate '"$RATELIMIT"
+}
+
 
 # Parses password to var or exits
 loadPassword() {
@@ -389,7 +435,7 @@ isPiped() {
         if [ "$1" = '-' ] || [ "$1" = '.' ] ; then
                 return $TRUE
         fi
-        return $FALSE
+        return $FALSE        
 }
 
 
@@ -419,6 +465,25 @@ isGlobbing() {
 }
 
 
+# If we should abort when curl returns a XML response, return $TRUE, else $FALSE
+abortOnDavErrors() {
+        return $ABORTONERRORS
+}
+
+
+# If have a dav error, return $TRUE, else $FALSE
+hasDavErrors() {
+        isEmpty "$CURLRESPONSES" && return $FALSE
+        return $TRUE
+}
+
+
+# If have a dav error, return $TRUE, else $FALSE
+checkAbort() {
+        abortOnDavErrors && hasDavErrors && logResult
+}
+
+
 
 
 
@@ -441,12 +506,12 @@ rawUrlEncode() {
                 esac
                 encoded+="${o}"
         done
-        echo "${encoded}"    # You can either set a return variable (FASTER)
+        echo "${encoded}"    # You can either set a return variable (FASTER) 
         REPLY="${encoded}"   #+or echo the result (EASIER)... or both... :p
 }
 
-
-escapeWhitespaces() {
+# Escape specific chars needed
+escapeChars() {
         local string="${1}"
         local strlen=${#string}
         local encoded=""
@@ -456,16 +521,23 @@ escapeWhitespaces() {
                 c=${string:$pos:1}
                 case "$c" in
                         " " ) o='%20' ;;
+                        '#' ) o='%23' ;;
                         * ) o="${c}" ;;
                 esac
                 encoded+="${o}"
         done
-        echo "${encoded}"    # You can either set a return variable (FASTER)
-        REPLY="${encoded}"   #+or echo the result (EASIER)... or both... :p
+        echo "${encoded}"    # You can either set a return variable (FASTER) 
+        #REPLY="${encoded}"   #+or echo the result (EASIER)... or both... :p
 
 }
 
 
+# Decode '%2F' into '/'
+decodeSlash() {
+       isEmpty "$1" && return 9
+       echo "$(echo "$1" | sed 's/\%2f/\//g' | sed 's/\%2F/\//g')"
+}
+
 
 
 
@@ -480,24 +552,25 @@ createDir() {
         isEmpty "$1" && initError 'Error! Cannot create folder with empty name.'
         getScreenSize
         logSameLine "$1 > "
-        eout="$(escapeWhitespaces "$1")"
-        #echo "$CURLBIN"$INSECURE --silent -X MKCOL -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX/$1"
-        #"$CURLBIN"$INSECURE --silent -X MKCOL -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX/$eout" | cat ; test ${PIPESTATUS[0]} -eq 0
+        eout="$(escapeChars "$1")"
+        #echo "$CURLBIN"$INSECURE --silent -X MKCOL -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX$INNERPATH/$1" 
         cstat="$(createDirRun "$eout" 2>&1)"
         #echo " -- $cstat"
         if ! isEmpty "$cstat"; then
+                curlAddResponse "$cstat" "Send Folder: \"$eout\""
                 msg="$(echo "$cstat" | grep '<s:message>' | sed -e 's/<[^>]*>//g' -e 's/^[[:space:]]*//')"
                 isEmpty "$msg" && msg="$(echo "$cstat" | grep '<s:exception>' | sed -e 's/<[^>]*>//g' -e 's/^[[:space:]]*//')"
                 log "$msg"
         else
                 log 'OK'
         fi
+        checkAbort # exits if DAV errors AND not ignoring them
 }
 
 
 # Create a directory with -X MKCOL
 createDirRun() {
-        "$CURLBIN"$INSECURE --silent -X MKCOL -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX/$1" | cat ; test ${PIPESTATUS[0]} -eq 0
+        "$CURLBIN"$INSECURE --silent -X MKCOL -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX$INNERPATH/$1" | cat ; test ${PIPESTATUS[0]} -eq 0
         ecode=$?
         curlAddExitCode $ecode
         return $ecode
@@ -521,7 +594,7 @@ sendDir() {
         # MacOS / BSD readlink does not have the -f option
         # Get bash implementation from pdfScale.sh if needed
         # For now PWD seems to be enough
-        if [[ "$fbn" == '.' ]]; then
+        if [[ "$fbn" == '.' ]]; then 
                 fbn="$PWD"
                 fbn="$("$BASENAMEBIN" "$fbn")"
         fi
@@ -530,7 +603,7 @@ sendDir() {
 
         # Create main/root folder that is being sent
         createDir "$fbn"
-
+        
         # Create whole directory tree at destination
         for d in "${DIRLIST[@]}"; do
                 if ! isEmpty "$d"; then
@@ -539,9 +612,9 @@ sendDir() {
         done
 
         log $'\n'"SENDING ALL FILES FROM FOLDER TREE"$'\n'"=================================="$'\n'
-
+        
         # Send all files to their destinations
-        for f in "${FILELIST[@]}"; do
+        for f in "${FILELIST[@]}"; do 
                 if ! isEmpty "$f"; then
                         OUTFILE="$fbn/$f"
                         log "$OUTFILE > "
@@ -558,12 +631,16 @@ logResult() {
         local fileString="$("$BASENAMEBIN" "$FILENAME")"
         isRenaming && fileString="$("$BASENAMEBIN" "$FILENAME") (renamed as $OUTFILE)"
         log $'\n'"SUMMARY"$'\n'"======="$'\n'
-
+                  
         if [ $CURLEXIT -eq 0 ]; then
-                log " > All Curl calls exited without errors"$'\n'" > Attempt to send completed > $fileString"
+                if isEmpty "$CURLRESPONSES"; then
+                        log " > All Curl calls exited without errors and no WebDAV errors were detected"$'\n'" > Attempt to send completed > $fileString"
+                else
+                        log " > All Curl calls exited without errors, but webdav errors"$'\n'"   were detected while trying to send $fileString"$'\n\n'"Curl Log:"$'\n'"$CURLRESPONSES"
+                fi
                 exit 0
         fi
-        log " > Curl errors detected when sending > $fileString"$'\n'" > Summed Curl exit codes: $CURLEXIT"
+        log " > Curl execution errors were detected when sending > $fileString"$'\n'" > Summed Curl exit codes: $CURLEXIT"
         exit $CURLEXIT
 }
 
@@ -575,13 +652,17 @@ sendFile() {
         elif isEmpty "$OUTFILE"; then # If we are not renaming, use the input file name
                 OUTFILE="$("$BASENAMEBIN" "$1")"
         fi
-
+        
         getScreenSize
-        eout="$(escapeWhitespaces "$OUTFILE")"
+        eout="$(escapeChars "$OUTFILE")"
         # Send file
-        #echo "$CURLBIN"$INSECURE$VERBOSE -T "$1" -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX/$eout"
-        "$CURLBIN"$INSECURE$VERBOSE$GLOBCMD -f -T "$1" -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX/$eout" | cat ; test ${PIPESTATUS[0]} -eq 0
-        curlAddExitCode $?
+        #echo "$CURLBIN"$INSECURE$VERBOSE -T \""$1"\" -u \""$FOLDERTOKEN":"$PASSWORD"\" -H \""$HEADER"\" \""$CLOUDURL/$PUBSUFFIX$INNERPATH/$eout"\"
+        #"$CURLBIN"$LIMITCMD$INSECURE$VERBOSE$GLOBCMD -T "$1" -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX$INNERPATH/$eout" | cat ; test ${PIPESTATUS[0]} -eq 0
+        resp="$("$CURLBIN"$LIMITCMD$INSECURE$VERBOSE$GLOBCMD -T "$1" -u "$FOLDERTOKEN":"$PASSWORD" -H "$HEADER" "$CLOUDURL/$PUBSUFFIX$INNERPATH/$eout")"
+        stat=$?
+        curlAddResponse "$resp" "Send File: \"$eout\""
+        curlAddExitCode $stat
+        checkAbort # exits if DAV errors AND not ignoring them
 }
```
Can you confirm that the script you have added by this recipe works in DTS?

PLangowski replied:
this is outdated, this is the current commit with patch: 351e05eeeda8dc3e432033efc4d5a39993808937

DaniilKl replied:
Ok, but you still import version `2.2.8`, could you please confirm that it works in DTS?

PLangowski replied:
@DaniilKl 
Sending file to `$CLOUDSENS_URL`:

```
SENDING SINGLE FILE
===================

testfile >
######################################################################## 100.0%

SUMMARY
=======

 > All Curl calls exited without errors and no WebDAV errors were detected
 > Attempt to send completed > testfile
```

Sending a file to a non-existing share on our cloud:

```
SENDING SINGLE FILE
===================

testfile >
######################################################################## 100.0%curl: (22) The requested URL returned error: 401


SUMMARY
=======

 > Curl execution errors were detected when sending > testfile
 > Summed Curl exit codes: 22
```

We have both the status code and return code so it seems ok to me.

DaniilKl replied:
Could you verify that new version of the script (which is 2.2.8 vs old one - 2.2.1) can download shares or upload HCL report?

PLangowski replied:
Cloudsend only sends files/folders.

From https://github.com/tavinus/cloudsend.sh?tab=readme-ov-file#tavinus-cloudsender2:
> Bash script that uses curl to send files and folders to a [nextcloud](https://nextcloud.com/) / [owncloud](https://owncloud.org/) publicly shared folder.

DaniilKl replied:
OK, my bad.

DaniilKl commented on scripts/update_components.sh:
```diff
@@ -90,7 +90,7 @@ do_update() {
         result=$((line_number - 2))
 
-        sed ""$result"a\* Updated $1 to revision $last_commit" $CHANGELOG_FILE > $CHANGELOG_TMP_FILE
+        sed "\"$result\"a\* Updated $1 to revision $last_commit" $CHANGELOG_FILE > $CHANGELOG_TMP_FILE
```
Testing results showed an error:
```bash
sed: -e expression #1, char 1: unknown command: `"'
```

```suggestion
        sed "$result a\* Updated $1 to revision $last_commit" $CHANGELOG_FILE > $CHANGELOG_TMP_FILE
```

PLangowski replied:
5661ca93e27a4b3d8c9f9e0b06bd197d274fbef1

PLangowski commented on meta-dts-distro/recipes-bsp/flashrom/flashrom_git.bb:
```diff
@@ -1,4 +1,4 @@
-DESCRIPTION = "flashrom is a utility for identifying, reading, writing, verifying and erasing flash chips"
+SUMMARY = "flashrom is a utility for identifying, reading, writing, verifying and erasing flash chips"
```
`utility` begins with a consonant actually. It's true that `u` is the first letter, but what matters is how the letter is pronounced.
https://www.dictionary.com/browse/utility#more-about

PLangowski commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,59 @@
+    rev: 5.3.2
+    hooks:
+      - id: oelint-adv
+        args: [--rulefile=.oelint-ruleset.json, --noinfo, --nowarn, --quiet]
```
I talked to @macpijan and we decided to do the following:
We will ignore info, but not warnings and I will change some checks' severity from warning to info so that they don't appear. I did that in this commit: 3b03cfee81def65b977e844d732577f7d9a79974

PLangowski commented on meta-dts-distro/recipes-bsp/coreboot-utils/intelp2m_git.bb:
```diff
@@ -1,5 +1,9 @@
+SUMMARY = "\
+convert the configuration DW0/1 registers value from an inteltool dump to \
+coreboot macros."
+
```
5661ca93e27a4b3d8c9f9e0b06bd197d274fbef1

PLangowski commented on meta-dts-distro/recipes-bsp/coreboot-utils/inteltool_git.bb:
```diff
@@ -1,5 +1,9 @@
+SUMMARY = "\
+Provides information about the Intel CPU/chipset hardware configuration \
+(register contents, MSRs, etc)."
+
```
5661ca93e27a4b3d8c9f9e0b06bd197d274fbef1

PLangowski replied:
@DaniilKl We've decided to enable the default `oelint.vars.multilineident` rule. It actually requires the line break right after `"` if we're dealing with multiline variable assignments. I will change this now.

PLangowski commented on meta-dts-distro/recipes-extended/hw-probe/hw-probe_1.6.5.bb:
```diff
@@ -1,9 +1,11 @@
-DESCRIPTION = "tool to probe for hardware, check operability and find drivers"
+SUMMARY = "tool to probe for hardware, check operability and find drivers"
```
5661ca93e27a4b3d8c9f9e0b06bd197d274fbef1

PLangowski commented:
@DaniilKl @macpijan I added the `fix` flag so that oelint fixes some of the files (doesn't work on all checks).
I also added a custom rule that will check for multiline variable assignment and require a 4-space indent to be present.  3b03cfee81def65b977e844d732577f7d9a79974 This, however, does not work yet. I will try to work this out.

PLangowski commented:
The rule file could not be loaded because it was in a hidden directory and Python couldn't find it.

PLangowski commented:
@macpijan I got my custom rule to work. It turns out that some of the existing recipes comply to the original oelint rule - They indent to the `"` character in the first line:
```
SRC_URI = "git://github.com/nix-community/acpi_call.git;branch=master;protocol=https \
           file://0001-Makefile-fixing-build-errrors.patch \
           file://0001-GPLv3-licence-file-added.patch \
           "
```

We need to decide what we want: Always a 4 space indent or indenting up to `"`? What do you think?

macpijan commented:
> We need to decide what we want: Always a 4 space indent or indenting up to "? What do you think?

Which rule is that? This one as linked in the logs here: https://github.com/Dasharo/meta-dts/pull/129#discussion_r1634513475

Can it be autofixed by the tool?

I have been using so far the 4 spaces, but there may be some reason why the upstream linter tool uses the other approach. 

It was enough for 3mins search to find this: https://docs.yoctoproject.org/dev/contributor-guide/recipe-style-guide.html#variable-formatting 

This should close the discussion and our preference does not matter much.

macpijan commented on .oelint-ruleset.json:
```diff
@@ -0,0 +1,249 @@
+  "oelint.var.nativefilename": "warning",
+  "oelint.var.nativesdkfilename": "warning",
+  "oelint.var.order": "info",
+  "oelint.var.order.SUMMARY": "info",
```
Is it really that bad to enable it? Why not follow the rules of style guide from the upstream project: https://docs.yoctoproject.org/dev/contributor-guide/recipe-style-guide.html#recipe-ordering
IMHO, we should skip them only when there is a good reason to.

Avoiding following the rules, will just make the possible upstreaming unnecessarily more difficult task, so we will do it even lass then we do today.

PLangowski replied:
Done https://github.com/Dasharo/meta-dts/pull/129/commits/80542fdfc24abbcd050ea3e3ead7f7522d866b90#diff-cceeba45d98af3b36ef0a37e8ffdda5ff5d1ba0084c4ad21dad516a28f2e20a4R201-R228

PLangowski commented:
Unfortunately, a lot of the rules, including the indent, do not have an autofix and need to be changed manually

macpijan commented on meta-dts-distro/recipes-tests/converged-security-suite/bg-suite_2.7.0.bb:
```diff
@@ -1,6 +1,6 @@
-FILESEXTRAPATHS += "${THISDIR}/${PN}"
+FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
```
I love that such obvious mistakes are now detected automatically by the tool.
But in this particular case, we do not need this line at all - it is required in bbappend files only.

macpijan commented on meta-dts-distro/recipes-support/mei-amt-check/files/0001-Makefile-removed-setting-CC-variable.patch:
```diff
@@ -15,8 +15,7 @@ index c98d181..e1700dc 100644
+
  all: $(PROGS)
--- 
+--
```
Modification of patches may break them, we can consider skipping some rules for patch files. @DaniilKl had some ideas on this IIRC.

m-iwanicki replied:
@PLangowski To exclude `.patch` files from being `fixed` by this hook

```suggestion
      - id: trailing-whitespace
        exclude: .*\.patch$
```

You can also add `exclude` at the top level to exclude patches from being checked by pre-commit at all

PLangowski replied:
I restored the patches to their original form https://github.com/Dasharo/meta-dts/pull/129/commits/8725595a288f981ff2a1a8b78e397d015b39c149

DaniilKl commented on meta-dts-distro/recipes-dts/images/dts-base-image.inc:
```diff
@@ -1,4 +1,4 @@
-DESCRIPTION = "dts image"
+SUMMARY = "dts image"
```
@PLangowski, missed this one?

DaniilKl replied:
Was fixed here 5661ca93e27a4b3d8c9f9e0b06bd197d274fbef1.

DaniilKl commented on meta-dts-distro/recipes-dasharo/3mdeb-secpack/3mdeb-secpack_git.bb:
```diff
@@ -38,7 +39,6 @@ do_install() {
-FILES:${PN} = " \
+FILES:${PN} += " \
     ${ROOT_HOME} \
-    ${sysconfdir} \
```
@PLangowski, why this one have been deleted? Is this directory added to `FILES` by environment?

PLangowski replied:
Yes, it's added by poky by default https://git.yoctoproject.org/poky/tree/meta/conf/bitbake.conf#n332

DaniilKl commented on meta-dts-distro/recipes-support/cloudsend/cloudsend_git.bb:
```diff
@@ -0,0 +1,26 @@
+SUMMARY = "Bash script that uses curl to send files and folders to a nextcloud / owncloud publicly shared folder."
```
```suggestion
SUMMARY = "Bash script that uses curl to send files and folders to a nextcloud /
    owncloud publicly shared folder."
```

PLangowski replied:
d9dea8d9b86629b097f6d3bbd8f01aee7df2e817

DaniilKl commented on meta-dts-distro/recipes-support/dasharo-ectool/dasharo-ectool_0.3.8.bb:
```diff
@@ -1,13 +1,21 @@
+
+DEPENDS += "hidapi"
 PV:append = ".AUTOINC+411fab9b7c"
+SRC_URI:append = " gitsm://github.com/Dasharo/ec.git;protocol=https;nobranch=1"
```
Why append here? It is first time `SRC_URI` is being set.

PLangowski replied:
I'm not sure - the original recipe used the `+=` operator here. Maybe you're right that it's better to just use `=` here. d9dea8d9b86629b097f6d3bbd8f01aee7df2e817

DaniilKl commented on meta-dts-distro/recipes-bsp/coreboot-utils/futil_git.bb:
```diff
@@ -1,29 +1,33 @@
     flashrom \
 "
 
+SRC_URI:append = " \
```
Why append here? It is first time `SRC_URI` being set.

PLangowski replied:
d9dea8d9b86629b097f6d3bbd8f01aee7df2e817

PLangowski commented:
@macpijan I managed to create an autofix for the indent check. I seems to work except for one case: When the first line of the variable alrerady has something other than `"\`, e.g.:

```
SRC_URI = "file://some-file \
<other files>
```

I haven't managed to get the metod to fix that.

macpijan commented:
Sounds like a good start @PLangowski , nice work.

macpijan commented:
One more: let's try to cleanup commits.

PLangowski commented:
I cleaned up the commits

DaniilKl commented:
@PLangowski, ping me please when review will be needed and use `draft` for a PR next time to signal when the PR is not ready for review.

PLangowski commented:
@DaniilKl it is ready for review now.

macpijan commented:
Good work overall @PLangowski I have very high hopes on overall quality increase because of these changes.
